pandas/core/indexes/datetimelike.py

```diff
@@ -692,6 +693,9 @@ def __add__(self, other):
                 return self._add_datelike(other)
             elif isinstance(other, Index):
                 return self._add_datelike(other)
+            elif is_integer_dtype(other) and self.freq is None:
+                # GH#19123
+                raise NullFrequencyError("Cannot shift with no freq")
```

Comment:
just make this a ValueError, we don't want to have custom error messages normally.

Reply:
We need to catch this specifically in core.ops.  Want to catch by checking the error message?

Reply:
>  don't want to have custom error messages normally

The error message here "Cannot shift with no freq" is the same as the error message raised if adding just an integer when `self.freq` is None.